atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/KvsSweepQueueWriter.java

```diff
@@ -52,17 +53,35 @@ public KvsSweepQueueWriter(KeyValueService kvs, TableReference tableRef, WriteIn
 
     @Override
     public void enqueue(List<WriteInfo> allWrites) {
+        Map<Cell, byte[]> referencesToDedicatedCells = new HashMap<>();
         Map<Cell, byte[]> cellsToWrite = new HashMap<>();
         Map<PartitionInfo, List<WriteInfo>> partitionedWrites = partitioner.filterAndPartition(allWrites);
-        partitionedWrites.forEach((partitionInfo, writes) -> cellsToWrite.putAll(populateCells(partitionInfo, writes)));
-        if (!cellsToWrite.isEmpty()) {
-            kvs.put(tableRef, cellsToWrite, SweepQueueUtils.WRITE_TS);
-        }
+
+        partitionedWrites.forEach((partitionInfo, writes) -> {
+            referencesToDedicatedCells.putAll(populateReferences(partitionInfo, writes));
+            cellsToWrite.putAll(populateCells(partitionInfo, writes));
+        });
+
+        partitionedWrites.keySet().stream()
+                .map(PartitionInfo::timestamp)
+                .mapToLong(x -> x)
+                .max()
+                .ifPresent(timestamp -> {
+                    write(referencesToDedicatedCells, timestamp);
+                    write(cellsToWrite, timestamp);
+                });
```

Comment:
`max()` seems unnecessary given that right now each enqueue strictly consists of writes at the same timestamp, but post offline discussion there may be plans to do something different or cleverer later on.

Also: we may want to revisit this much later along the FFT project, if we're trying to make things faster. In particular, it looks like we now call multiPut twice instead of once.

Reply:
For the first point, you're right that all the timestamps should be the same. Maybe it's a bit of too defensive coding, but the contract of enqueue does not specify that all the write infos must have the same start timestmap.For the second point, this is because we don't want to write dedicated cells and then explode without writing the references, which can happen depending on how the writes get batched internally.